gui/biomes.lua

```diff
@@ -0,0 +1,379 @@
+                table.insert(biomeList, {biome = biome, char = ch, typeId = biomeTypeId})
+                ix = #biomeList
+            end
+
+            biomesZY[x] = ix
+
+            ::continue::
+        end
+    end
+    --end
+end
+
+-- always gather info at the very bottom first: this ensures the important biomes are
+-- always in the same order (high up in the air strange things happen)
+gatherBiomeInfo(0)
+
+--------------------------------------------------------------------------------
+
+local TITLE = "Biomes"
+
+if RELOAD then BiomeVisualizerLegend = nil end
+BiomeVisualizerLegend = defclass(BiomeVisualizerLegend, widgets.Window)
+BiomeVisualizerLegend.ATTRS {
+    frame_title=TITLE,
+    frame_inset=0,
+    resizable=true,
+    resize_min={h=5, w = 14},
+    frame = {
+        w = 47,
+        h = INITIAL_LIST_HEIGHT + 2 + INITIAL_INFO_HEIGHT,
+        -- just under the minimap:
+        r = 2,
+        t = 18,
+    },
+}
+
+local function GetBiomeName(biome, biomeTypeId)
+    -- based on probe.cpp
+    local sav = biome.savagery
+    local evi = biome.evilness;
+    local sindex = sav > 65 and 2 or sav < 33 and 0 or 1
+    local eindex = evi > 65 and 2 or evi < 33 and 0 or 1
+    local surr = sindex + eindex * 3 +1; --in Lua arrays are 1-based
+
+    local surroundings = {
+        "Serene", "Mirthful", "Joyous Wilds",
+        "Calm", "Wilderness", "Untamed Wilds",
+        "Sinister", "Haunted", "Terrifying"
+    }
+    local surrounding = surroundings[surr]
+
+    local name = biomeTypeNames[df.biome_type[biomeTypeId]] or "DFHACK_Unknown"
+
+    return ([[%s %s]]):format(surrounding, name)
+end
+
+function BiomeVisualizerLegend:init()
+    local list = widgets.List{
+        view_id = 'list',
+        frame = { t = 0, b = INITIAL_INFO_HEIGHT + 1 },
+        frame_inset = 0,
+        icon_width = 1,
+        text_pen = { fg = COLOR_GREY, bg = COLOR_BLACK }, -- this makes selection stand out more
+        on_select = self:callback('onSelectEntry'),
+    }
+    local tooltip_panel = widgets.Panel{
+        view_id='tooltip_panel',
+        autoarrange_subviews=true,
+        frame = { b = 0, h = INITIAL_INFO_HEIGHT },
+        frame_style=gui.INTERIOR_FRAME,
+        frame_background=gui.CLEAR_PEN,
+        subviews={
+            widgets.Label{
+                view_id='label',
+                text_to_wrap='',
+                scroll_keys={},
+            },
+        },
+    }
+    self:addviews{
+        list,
+        tooltip_panel,
+    }
+
+    self.list = list
+    self.tooltip_panel = tooltip_panel
+
+    self:UpdateChoices()
+end
+
+local PEN_ACTIVE_ICON = dfhack.pen.parse{tile=TILE_HIGHLIGHTED}
+local PEN_NO_ICON = nil
+
+function BiomeVisualizerLegend:get_icon_pen_callback(ix)
+    return function ()
+        if self.SelectedIndex == ix then
+            return PEN_ACTIVE_ICON
+        else
+            return PEN_NO_ICON
+        end
+    end
+end
+
+function BiomeVisualizerLegend:onSelectEntry(idx, option)
```

Comment:
in my testing, `option` can be nil. it might be a good idea to check that and return early here if option is nil

Reply:
I've added a check in the `get_tooltip_text(option)`